modules/e2ee/OlmAdapter.js

```diff
@@ -470,6 +572,199 @@ export class OlmAdapter extends Listenable {
             }
             break;
         }
+        case OLM_MESSAGE_TYPES.SAS_START: {
+            if (olmData.session) {
+                if (olmData.sasVerification?.sas) {
+                    logger.warn(`SAS already created for participant ${pId}`);
+
+                    return;
+                }
+
+                const { uuid } = msg.data;
+
+                olmData.sasVerification.sas = new Olm.SAS();
+                olmData.sasVerification.uuid = uuid;
+
+                const pubKey = olmData.sasVerification.sas.get_pubkey();
+                const olmUtil = new Olm.Utility();
+                const commitment = olmUtil.sha256(pubKey + msg.data);
+
+                olmUtil.free();
+
+                /* The first phase of the verification process, the Key agreement phase
+                   https://spec.matrix.org/latest/client-server-api/#short-authentication-string-sas-verification
+                */
+                const ack = {
+                    [JITSI_MEET_MUC_TYPE]: OLM_MESSAGE_TYPE,
+                    olm: {
+                        type: OLM_MESSAGE_TYPES.SAS_ACCEPT,
+                        data: {
+                            uuid,
```

Comment:
Can we rename this uuid to `transaction_id` everywhere in the SAS part? It better matches the Matrix spec.